src/spu/spu.cc

```diff
@@ -907,8 +882,12 @@ void PCSX::SPU::impl::SetupThread() {
     bEndThread = 0;  // init thread vars
     bThreadEnded = 0;
     bSpuInit = 1;  // flag: we are inited
-
-    hMainThread = std::thread([this]() { MainThread(); });
+    // Threads are expensive and causes Thread init and switch overheads
+    // hMainThread = std::thread([this]() { MainThread(); });
+    main_t_rst = std::async([this]() {
```

Comment:
If anything, std::async will also set up a thread to function properly. This only obfuscates the fact.

Reply:
I made `main_t_rst` to make it run concurrently. otherwise, it will run sequentially which Emu will hang. because the future is missed when it's not global. 

It's better to roll it back to previous code, you are fine